### PR TITLE
Fix batch ops windows

### PR DESCRIPTION
**Pull request by jimkeir**
There are two problems with macro operations on Windows. First, the file dialog doesn't handle many filenames, it returns a corrupt path. There needs to be special handling for buffer overflow to handle the first two bytes of the return buffer being used as the required buffer size, not text. Second, a bug in libsndfile left file handles open meaning that the batch process would fail after a varying number of files - around 8000 in my case, depending on the version of Windows. I can submit this separately to libsndfile if you prefer.

### CODE REVIEW

File: lib-src/libsndfile/src/file_io.c [hunk 1]
```diff
@@ -627,10 +627,20 @@ psf_fclose (SF_PRIVATE *psf)
 		return 0 ;
 		} ;
 
-	if ((retval = psf_close_handle (psf->file.handle)) == -1)
+	if (psf->file.filedes != -1)
+	{
+		retval = close(psf->file.filedes);
```
Comment (Paul-Licameli) [inline lib-src/libsndfile/src/file_io.c:632]: Minor point, leave space before ( to respect the conventions in this library.

File: lib-src/libsndfile/src/common.h [hunk 2]
```diff
@@ -354,10 +354,9 @@ typedef struct
 	void 			*handle, *hsaved ;
 
 	int				use_wchar ;
-#else
+#endif
 	/* These fields can only be used in src/file_io.c. */
 	int 			filedes, savedes ;
```
Comment (Paul-Licameli) [inline lib-src/libsndfile/src/common.h:359]: You make no use of savedes.  So the more minimal change logically (though not textually) should leave savedes out of conditional compilation for Windows.

**Comment by Paul-Licameli** [comment]
Thanks for the bug fix in libsndfile.  I made careful review and I think more is needed to complete the fix in all cases.  I have not yet attempted review of the other commit.

Perhaps these two commits should be in separate pull requests.  If libsndfile really has a resource leak bug on Windows, it stands by itself as a good bug fix perhaps for other uses cases too.

**Comment by LWinterberg** [comment]
Hello! We just [switched master to 4.0.0](https://forum.audacityteam.org/t/master-has-been-switched-to-4-0-0-development/120444) which makes this PR incompatible and/or obsolete as we're switching UI framework and reorganizing large parts of the codebase. You're very welcome to make these kinds of contributions in the future though, after things stabilize a little bit.

Also, sorry for this comically large delay.

**Pull request closed without merging**